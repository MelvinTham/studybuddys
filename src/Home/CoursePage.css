.avatar {
    display: flex;
    justify-content: center;
}

.Calendar{
    padding: 45px 70px;
    display: block;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 10%;
}

.browserTitle{
    background-attachment: fixed;
    background-position: center;
    background-size: cover;
    font-family: 'Inconsolata', monospace;
    font-size: 40px;
  }

  .nav {  
    width: 100vw; 
    z-index: 1 !important;
    color:white;
  }