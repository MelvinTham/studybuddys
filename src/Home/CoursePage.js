import React, { PureComponent } from 'react';
import {
    Typography,
    ExpansionPanel,
    ExpansionPanelSummary,
    ExpansionPanelDetails,
    TextField,
    Button,
} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import './CoursePage.css';
import axios from 'axios';
import * as firebase from 'firebase';
import Navbar from "./Navbar";
import Calendar from "./Calendar2";
// import Calendar from 'rc-calendar';
import LuxonUtils from '@date-io/luxon';
import { MuiPickersUtilsProvider } from 'material-ui-pickers';

class CoursePage extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            name:"",
            course: '',
            newQuestion: "",
            questions: [],
            response: {0: ["ASDSADASDA","asdad", "asdasda"], 1: ["ASDSADASDA", "jabababab"], 2: ["ASDSADASDA", "hhahahahah"], 3: ["ASDSADASDA", "O"], 4: ["ASDSADASDA", "M"], 5:["ASDSADASDA", "scoob", "G"]},
            people: ["Jack", "Andy", "Bob", "Pauline", "Luis", "Connie", "Sponge"],
            calendarIsOpen: false
        }
        this.handleChange = this.handleChange.bind(this);
        this.createQuestion = this.createQuestion.bind(this);   
        this.logout = this.logout.bind(this);
        this.getUserName = this.getUserName.bind(this);
        this.getQuestions = this.getQuestions.bind(this);
        this.openCalendar = this.openCalendar.bind(this);
    }

    componentDidMount() {
        const { courseName } = this.props.match.params
        fetch(`/course/${courseName}`).then(this.setState({ course: courseName }));
        this.getUserName();
        this.getQuestions();
    }

    handleChange = name => event => {
        this.setState({
            [name]: event.target.value,
        });
    };

    logout(e){
        e.preventDefault();
        firebase.auth().signOut();
        this.props.history.push('/');
    }

    getQuestions() {
        var course = {
            courseName: this.state.course
        };
        console.log("hi " + this.state.course)
        axios.post('https://triple-bonito-221722.appspot.com/api/getQuestions', course)
            .then(response => {
                this.setState({questions: response.data.questions})
                console.log(response.data);
            })
    }

    getUserName(e){
        var page = this;
        firebase.auth().onAuthStateChanged(function(user) {
        if (user) {
            var info = {
                id: user.uid
            }
            axios.post('https://triple-bonito-221722.appspot.com/api/getUsername', info)
            .then(response => {
                page.setState({name : response.data.name})
            })
        }
        });
    }

    createQuestion() {
        var course = this.state.course;
        var newQuestion = this.state.newQuestion;
        firebase.auth().onAuthStateChanged(function(user){
            if(user) {
                var question = {
                    id: user.uid,
                    courseName: course,
                    userQuestion: newQuestion
                };
                axios.post('https://triple-bonito-221722.appspot.com/api/createQuestion', question)
                console.log("course is " + course)
                console.log("question is: " + newQuestion)
                console.log(question)
            }
        })
        this.getQuestions();
    }

    openCalendar() {
        this.setState({
            calendarIsOpen: !this.state.calendarIsOpen
        })
    }

    render() {

        return (
           
            <div data-aos ="fade-in" data-aos-easing="linear" data-aos-duration="800" style = {{display: "flex", flexDirection: "column"}}>
               <div>
                    <div style = {{float: "right", display: "inline-block"}}>
                        <span>{this.state.name}</span>
                        <Button onClick={this.logout}>Logout</Button>
                    </div>
                     <Navbar/>
             </div>
                {<Typography variant = "h1" style = {{margin: "16px auto"}}>{this.state.course}</Typography>}
                <Button 
                    className="Calendar"
                    type="submit"
                    onClick={this.openCalendar}>
                    Meet Up
                    </Button>
                {
                    this.state.calendarIsOpen
                    ?
                    <MuiPickersUtilsProvider 
                        utils={LuxonUtils}>
                        <Calendar />
                    </MuiPickersUtilsProvider>
                    : null
                }
                <div className = "flexCenter">
                    {this.state.questions.map((data, key) => {
                    return (
                        <ExpansionPanel key = {key} style = {{width: "70%"}}>
                            <ExpansionPanelSummary 
                                expandIcon={<ExpandMoreIcon/>} 
                                style = {{borderBottom: "1px solid black"}}
                            >
                                <Typography>
                                    {data}
                                </Typography>
                            </ExpansionPanelSummary>
                            {this.state.response[key].map((ans, index) => {
                                return (
                                    <ExpansionPanelDetails key = {index}>
                                        <Typography>
                                            {ans}
                                        </Typography>
                                    </ExpansionPanelDetails>
                                )
                            })}
                            <TextField variant = "filled" multiline = {true} label = "Answer" fullWidth></TextField>
                            <Button fullWidth = {true}>Submit</Button>
                        </ExpansionPanel>
                    )}
                )}
                </div>

                <div style = {{margin: "15px 0"}}>
                    <TextField 
                        variant = "outlined"
                        multiline = {true} 
                        label = "Ask a Question"
                        onChange = {this.handleChange("newQuestion")}
                    >
                    </TextField>
                    <br/>
                    <Button 
                        type = "submit" 
                        variant = "contained"
                        onClick = {this.createQuestion}
                        style = {{margin: "15px 0"}}
                    >
                        Ask Away
                    </Button>
                </div>

                {/* TO DO  <div>
                    {this.state.people.map((data, key) => {
                        return (
                            <Avatar key = {key} style = {{width: "60px", height: "60px", display: "flex", justifyContent: "center"}}>{data}</Avatar>
                        )
                    })}
                </div>    */}
               
            </div>
        )
    }
}

export default CoursePage;