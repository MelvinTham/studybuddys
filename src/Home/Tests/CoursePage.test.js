import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import CoursePage from './../CoursePage';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('CoursePage component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <CoursePage />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <CoursePage />
        </Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
