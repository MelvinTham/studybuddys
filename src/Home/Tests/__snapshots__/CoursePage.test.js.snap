// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursePage component render snapshot 1`] = `
<div
  data-aos="fade-in"
  data-aos-duration="800"
  data-aos-easing="linear"
  style="display:flex;flex-direction:column"
>
  <div
    class="nav"
  >
    <div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="float:left"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-31 MuiSvgIcon-fontSizeLarge-39"
            focusable="false"
            role="presentation"
            style="color:white"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="flexRow"
      style="margin-left:auto"
    >
      <span
        style="font-size:22px;margin-right:10px"
      >
        <a
          href="/dashboard/profile"
          style="color:white"
        />
      </span>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="color:white;font-size:17px"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Logout
        </span>
      </button>
    </div>
  </div>
  <h1
    class="MuiTypography-root-54 MuiTypography-h1-66"
    style="margin:16px auto"
  />
  <div
    class="flexCenter"
    style="margin:30px;background-color:#ffffff;border:1px solid black;padding:20px;border-radius:10px;box-shadow:5px 5px 5px 5px #777777;-moz-box-shadow:0 0 10px #777777;-webkit-box-shadow:0 0 10px #777777"
  >
    <h4
      class="MuiTypography-root-54 MuiTypography-h4-69 MuiTypography-gutterBottom-81"
    >
      <u>
        Questions
      </u>
      :
    </h4>
    <div
      class="MuiFormControl-root-89"
      style="margin-top:20px;width:50%"
    >
      <label
        class="MuiFormLabel-root-104 MuiInputLabel-root-93 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-outlined-103"
        data-shrink="false"
      >
        Ask a Question
      </label>
      <div
        class="MuiInputBase-root-124 MuiOutlinedInput-root-111 MuiInputBase-formControl-125 MuiInputBase-multiline-132 MuiOutlinedInput-multiline-117"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-141 MuiOutlinedInput-notchedOutline-118"
          style="padding-left:8px"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-142"
            style="width:0.01px"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
        <div
          class="MuiPrivateTextarea-root-143"
        >
          <textarea
            aria-hidden="true"
            class="MuiPrivateTextarea-textarea-144 MuiPrivateTextarea-shadow-145"
            readonly=""
            rows="1"
            tabindex="-1"
          />
          <textarea
            aria-hidden="true"
            class="MuiPrivateTextarea-textarea-144 MuiPrivateTextarea-shadow-145"
            readonly=""
            rows="1"
            tabindex="-1"
          />
          <textarea
            aria-invalid="false"
            class="MuiPrivateTextarea-textarea-144 MuiInputBase-input-134 MuiOutlinedInput-input-119 MuiInputBase-inputMultiline-136 MuiOutlinedInput-inputMultiline-121"
            rows="1"
            style="height:19px"
          />
        </div>
      </div>
    </div>
    <br />
    <button
      class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-contained-13 MuiButton-raised-16"
      style="width:50%"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label-3"
      >
        Ask Away
      </span>
    </button>
  </div>
  <div
    class="MuiPaper-root-147 MuiPaper-elevation1-150 MuiPaper-rounded-148 MuiCard-root-146 flexRow"
    style="margin:10px auto;width:50%;height:50px"
  />
</div>
`;
